### ISSUE TITLE
Do proper management of EOSPAC tables on copy and "Finalize"

### ISSUE THREAD

**Issue by jhp-lanl** [lanl/singularity-eos#262]
The Finalize method needs to call `eos_DestroyTables()` on the tables that the object owns when `Finalize()` is called. Otherwise EOSPAC tables will not be destroyed and you could end up using a lot more memory.

Similarly, we should think about how to handle copies. If I understand the spiner EOS model, I think copies are shallow so calling `Finalize()` on one copy effectively gets rid of the memory for all other copies. I would propose doing the same thing for EOSPAC tables and then maybe handling/translating the `EOS_INVALID_TABLE_HANDLE` error that should be raised by any subsequent calls.

Tagging @Yurlungur and @dholladay00 for input

**Comment by Yurlungur** [comment]
Yes I agree with your suggestion @jhp-lanl that feels like the right model.

**Comment by Yurlungur** [comment]
This is mostly resolved, although we do not raise the `EOS_INVALID_TABLE_HANDLE` error.
